File: zerotierfix/src/main/java/net/kaaass/zerotierfix/service/Route.java
package kill.online.helper.zeroTier.service;

import kill.online.helper.zeroTier.util.InetAddressUtils;

import java.net.InetAddress;

import lombok.Data;

/**
 * 路由记录数据类
 */
@Data
public class Route {
    private final InetAddress address;
    private final int prefix;
    private InetAddress gateway = null;

    public boolean belongsToRoute(InetAddress inetAddress) {
        return this.address.equals(InetAddressUtils.addressToRoute(inetAddress, this.prefix));
    }
}

File: app/src/main/java/kill/online/helper/zeroTier/service/Route.kt
package kill.online.helper.zeroTier.service

import kill.online.helper.zeroTier.util.InetAddressUtils
import java.net.InetAddress

/**
 * 路由记录数据类
 */
class Route(
    val address: InetAddress? = null,
    private val prefix: Int = 0,
    var gateway: InetAddress? = null
) {
    fun belongsToRoute(inetAddress: InetAddress): Boolean {
        return address == InetAddressUtils.addressToRoute(inetAddress, prefix)
    }
}
